vivisect/analysis/generic/switchcase.py

```diff
@@ -96,8 +96,12 @@ def getSwitchBase(vw, op, vajmp, emu=None):
 
     # TODO: Want a more arch-independent way of doing this
     arrayOper = movOp.opers[1]
-    if not (isinstance(arrayOper, e_i386.i386SibOper) and (not (arrayOper.scale % 4 == 0))):
-        vw.verbprint("0x%x: Either arrayoper is not an i386SibOper or scale is wrong: %s" % (op.va, repr(arrayOper)))
+    if not isinstance(arrayOper, e_i386.i386SibOper):
+        vw.verbprint("0x%x: arrayOper is not an i386SibOper: %s" % (op.va, repr(arrayOper)))
+        return
+
+    if arrayOper.scale % 4 != 0:
+        vw.verbprint("0x%x: arrayoper scale is wrong: (%d mod 4 != 0)" % (op.va, arrayOper.scale))
```

Comment:
So before your most recent changes, we bailed if the class isn't a SibOper and the scale also wasn't a multiple of 4. Now we bail if either is true. Just want to confirm that we want to bail if either of those conditions are true and not just if both. You said "Right now I'm taking advantage of conditional shortcircuiting" - can you explain how exactly for my own benefit? :)

Reply:
Yup, we do. Every compiler I've seen defaults to using the SibOper to determine where in the jump table to bounce to, and scale only ever exists on the SibOper class, so we do want to check that the thing right here is a SibOper right before we deref scale off it it.

Oh, the conditional shortcircuting means that in a compound statement "if X and Y", X and Y are evaluated in order, so if X comes out to be False, it'll evaluate the entire statement as False. Similaryly, `if X or Y` would only evaluate X if X turned out to be true.

The not at the beginning made it a little convoluted as to when it'd get evaluated (since `not (X and Y)` ultimately means `not X or not Y`, so if X wasn't a SibOper it would shortcircuit to returning without grabbing the scale). So for clarity's sake it was better to change it. Functionally it should be the same.